# HG changeset patch
# User Rik <rik@octave.org>
# Date 1544770717 28800
#      Thu Dec 13 22:58:37 2018 -0800
# Node ID 59bc720d1bf90315760577744ee50d78d9ca6b31
# Parent  90487d5431cc9b1fa98d445c4eab7eaf7517b8d7
Fix scatter plots with a mix of float and integer inputs (bug #41166).

* __scatter__.m: During input validation cast mixed data of floating point and
integer to floating point.

diff --git a/scripts/plot/draw/private/__scatter__.m b/scripts/plot/draw/private/__scatter__.m
--- a/scripts/plot/draw/private/__scatter__.m
+++ b/scripts/plot/draw/private/__scatter__.m
@@ -31,16 +31,35 @@ function hg = __scatter__ (varargin)
 
   if (nd == 2)
     istart = 6;
   else
     z = varargin{6}(:);
     istart = 7;
   endif
 
+  ## Force mixtures of int and float data to be float (Bug #4116).
+  if (xor (isfloat (x), isfloat (y)))
+    if (isfloat (x))
+      y = cast (y, class (x));
+    else
+      x = cast (x, class (y));
+    endif
+  endif
+  if (nd != 2)
+    if (xor (isfloat (x), isfloat (z)))
+      if (isfloat (x))
+        z = cast (z, class (x));
+      else
+        x = cast (x, class (z));
+        y = cast (y, class (z));
+      endif
+    endif
+  endif
+
   if (istart <= nargin)
     s = varargin{istart}(:);
     if (isempty (s) || ischar (s))
       s = 36;
     endif
     if (! ischar (varargin{istart}))
       istart += 1;
     endif
